pyproject.toml

```diff
@@ -41,6 +41,7 @@ ignore = [
     "W605",
     "E701", # Multiple statements on one line
     "B006", # Do not use mutable data structures for argument defaults
+    "B019", # lru_cache and cache are used deliberately
```

Comment:
Maybe add `noqa: B019` to the deliberate lines, such that a future wrong cache is not used.

Reply:
It may be misplaced trust in myself or others, but I think I trust that developers will recognize that a cache leaks memory.

Reply:
I'm not as afraid of the memory leak (that is expected), but the incorrect use of it for method, like this, will print 3 times:

``` python
from functools import lru_cache

class T:

    @lru_cache
    def t(self, n):
        print("printing t")

T().t(10)
T().t(10)
T().t(10)
```

https://docs.astral.sh/ruff/rules/cached-instance-method/

Reply:
Ah indeed, missed that somehow. Will open a new PR.